src/llvm/lib/Target/MSP430/AsmParser/MSP430AsmParser.cpp

```diff
@@ -481,7 +481,11 @@ bool MSP430AsmParser::ParseOperand(OperandVector &Operands) {
         getLexer().Lex(); // Eat '+'
         return false;
       }
-      Operands.push_back(MSP430Operand::CreateIndReg(RegNo, StartLoc, EndLoc));
+      if (Operands.size() > 1) // Emulate @rd in destination position as 0(rd)
+        Operands.push_back(MSP430Operand::CreateMem(RegNo,
```

Comment:
Can we (in separate patch) recognize memoperand with zero offset in the asmprinter and print as @Reg instead of 0(Reg)? Could be a bit better for readability.

Reply:
Okay, I'll take a look. Thanks!